docs/source/entities.mdx

```diff
@@ -0,0 +1,321 @@
+---
+title: Entities in Apollo Federation
+sidebar_title: Entities (basics)
+description: Resolve types across multiple subgraphs
+---
+
+import { CodeColumns, ExpansionPanel } from 'gatsby-theme-apollo-docs';
+
+> ⚠️ Some details of entity behavior have changed in Federation 2. For a summary of these changes, [see what's new](/federation-2/new-in-federation-2/).
+
+In a federated graph, an **entity** is an object type that defines its fields across multiple subgraphs. Each subgraph is responsible for resolving whichever fields it defines for an entity.
+
+Here, the `Product` entity's fields are defined across two subgraphs:
+
+<CodeColumns>
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+```graphql
+# Inventory subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+Types _besides_ object types (unions, interfaces, etc.) _cannot_ be entities.
+
+## Defining an entity
+
+In a subgraph schema, you can designate any existing object type as an entity by adding the `@key` directive to its definition, like so:
+
+```graphql{2}
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+The `@key` directive defines the entity's **primary key**, which consists of one or more of the type's `fields`. In this example, the `Product` entity's primary key is its `id` field.
+
+**Every instance of an entity must be uniquely identifiable by its `@key` fields.** This is what enables your gateway to associate data from _different_ subgraphs with the _same_ entity instance.
+
+> An entity's `@key` cannot include fields that return a union or interface.
+
+## Referencing an entity
+
+After you define an entity in one subgraph, _other_ subgraphs can then reference that entity in their schema.
+
+Recall our example `Product` entity from the Products subgraph:
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+With this definition in place, the Reviews subgraph can then add a field of type `Product` to its `Review` type, like so:
+
+```graphql{3}
+# Reviews subgraph
+type Review {
+  product: Product!
+  score: Int!
+}
+
+# This is a required "stub" of the Product entity (see below)
+type Product {
+  id: ID!
```

Comment:
I'll note that as of preview, _if_ this subgraph is a fed2 subgraph (and see below, but I think we should try to have all of our example be of fed2 schema), the `id` would have to be marked `@shareable` here.

Which points to another consideration: how do we make it clear in the doc whether an example if of a fed2 schema or not? Should we systematically add the proper `@link` to clarify it? I think we might want to.

Reply:
Sounds good, I incorrectly assumed `id` would be auto-`@shareable` because it was a `@key` field _somewhere_ but I see it'd need to be a key field _here_.

And yes, we definitely need to add the `@link` snippet in lots of places, I'm thinking of doing a separate pass for that